### ISSUE TITLE
yarn cache dir fills up completely on cloud sandbox

### ISSUE THREAD

**Issue by Janpot** [codesandbox/codesandbox-client#7697]
# 🐛 bug report

## Preflight Checklist

- [x] I have read the [Contributing Guidelines][contributing] for this project.
- [x] I agree to follow the [Code of Conduct][code_of_conduct] that this project
      adheres to.
- [x] I have searched the issue tracker for an issue that matches the one I want
      to file, without success.

### To Reproduce

We run our examples in cloud sandbox. e.g.  https://codesandbox.io/p/sandbox/github/mui/mui-toolpad/tree/master/examples/custom-datagrid-column?file=%2Fpackage.json%3A1%2C1

Recently they started failing with `ENOSPC` errors. Upon investigation I noticed that the yarn cache directory reached 5GB. a yarn install on a clean cache folder should be only around 2.5GB. It looks like this directory never gets purged and only grows. As our examples evolve this inevitably breaks all codesandboxes based on them with a way to fix.

### Your Environment

Version: 3e519ff
Environment: production
Sandbox: 6yl6gb
Pitcher: 0.293.4
Manager: 1.23.2
Cluster: fc-eu-0

[code_of_conduct]: https://github.com/codesandbox/codesandbox-client/blob/master/CODE_OF_CONDUCT.md
[contributing]: https://github.com/codesandbox/codesandbox-client/blob/master/CONTRIBUTING.md

**Comment by frog-o** [comment]
I don't think this just effect sandboxes.  I have a similar problem with a repository I imported from github. My fork I been working with codesandbox-client keep wanting to install all dependency and fill everything up .  It would be nice if someone work on a way to improve what is cache and cash clean in general.    I also have problem with docker and found [this site](https://depot.dev/blog/docker-clear-cache) helpful in that case.  Right now  I find the cache and cache cleaning to be buggy.  Running some of the yarn clean command gave me error but seem to be working now.  It amusing how thing keep changing as they work on problems even though they don't seem to reply much (they seem to be read what people are reporting  though which is cool).  If this is what they are doing I don't mind and would like to say thanks.
